datalad/support/annexrepo.py

```diff
@@ -2060,7 +2060,7 @@ def _run_annex_command_json(self, command,
                 log_online=True
             ))
         # TODO: refactor to account for possible --batch ones
-        annex_options = ['--json']
+        annex_options = ['--json', '--json-error-messages']
```

Comment:
We would need to check from which version of annex it was introduced (I still have vague memory asking for it so it might have been not too long ago) and possibly boost our minimal annex version requirement accordingly.

Reply:
I thought we are fine: https://git-annex.branchable.com/todo/include_msg_with_possible_reason_why_command___40__e.g._add__41___failed_into_--json_output/